File: backend/src/models/games.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class GameCreate(BaseModel):
    session_id: UUID
    scores: list[int] = Field(min_length=1)


class GameItem(BaseModel):
    id: UUID
    session_id: UUID
    game_number: int
    score: int
    created_at: datetime


class GamesResponse(BaseModel):
    session_id: UUID
    games: list[GameItem]
    count: int
    average: float

File: frontend/src/types/games.ts
export type GameItem = {
  id: string;
  session_id: string;
  game_number: number;
  score: number;
  created_at: string;
};

export type GamesResponse = {
  session_id: string;
  games: GameItem[];
  count: number;
  average: number;
};

export type ScoreImportResult = {
  session_id: string;
  games_imported: number;
  scores: number[];
  errors: string[];
};

export type ScoreImportRequest = {
  csv_text: string;
  source: string;
};

export type FrameItem = {
  id: string;
  game_id: string;
  frame_number: number;
  ball1: number;
  ball2: number | null;
  ball3: number | null;
  is_strike: boolean;
  is_spare: boolean;
};

export type FramesResponse = {
  game_id: string;
  frames: FrameItem[];
};
